vignettes/r-markdown.Rmd

```diff
@@ -21,13 +21,17 @@ knitr::opts_chunk$set(
 ```
 
 R Markdown supports a variety of languages through the use of knitr language
-engines. One such engine is the `stan` engine, which allows users to write Stan
-programs directly in their R Markdown documents by setting the language of the
-chunk to `stan`.
+engines. Where users wish to write Stan programs as chunks directly in R Markdown documents there are three options: 
+1. the user wishes all the Stan chunks in the R Markdown document to be processed using RStan;  
+2. all Stan chunks are to be processed using CmdStanR; and
+3. some chunks are to be processed by RStan and some by CmdStanR. 
 
-Behind the scenes, the engine relies on RStan to compile the model code into an
-in-memory `stanmodel`, which is assigned to a variable with the name given by
-the `output.var` chunk option. For example:
+Behind the scenes in each option, the engine compiles the model code in each chunk into an
+in-memory model: a `stanmodel` if Rstan is being used, or a `CmdStanModel` in the CmdStanR case. The in-memory model is assigned to a variable with the name given by the `output.var` chunk option.
```

Comment:
One subtlety is that CmdStanR creates a compiled executable that technically isn't in-memory (from R's perspective). The CmdStanModel object itself is in-memory (so what you say here isn't wrong), but to run Stan it runs the compiled executable outside of R and reads the results back in (unlike RStan which does everything in-memory in R via Rcpp). To avoid any confusion about this maybe we should just remove "in-memory" and instead say something like the following, which I hope gets the idea across: 

> Behind the scenes in each option, the engine compiles the model code in each chunk and creates an object that provides methods to run the model: a `stanmodel` if Rstan is being used, or a `CmdStanModel` in the CmdStanR case. This model object is assigned to a variable with the name given by the `output.var` chunk option.

If you're ok with that change I can make it.

Reply:
I am entirely content with that change.